frontend/packages/operator-lifecycle-manager/src/components/clusterserviceversion.tsx

```diff
@@ -1038,28 +1038,54 @@ export const ClusterServiceVersionDetails: React.SFC<ClusterServiceVersionDetail
 
 export const CSVSubscription: React.FC<CSVSubscriptionProps> = ({
   obj,
-  subscriptions = [],
+  customData: { subscription },
   ...rest
 }) => {
   const EmptyMsg = () => (
     <MsgBox title="No Operator Subscription" detail="This Operator will not receive updates." />
   );
 
-  const subscription = React.useMemo(() => subscriptionForCSV(subscriptions, obj), [
-    obj,
-    subscriptions,
-  ]);
-
   return (
     <StatusBox EmptyMsg={EmptyMsg} loaded data={subscription}>
       <SubscriptionDetails {...rest} obj={subscription} clusterServiceVersions={[obj]} />
     </StatusBox>
   );
 };
 
+/**
+ * Give the name of the operator from subscription to the CSV Details Page
+ */
+export const CSVDetailsPageWrapper: React.FC<CSVDetailsPageWrapper> = (props) => {
+  const { match } = props;
+  const subscriptionResource = React.useMemo(
+    () => ({
+      isList: true,
+      kind: referenceForModel(SubscriptionModel),
+      namespace: match.params.ns,
```

Comment:
Is this right for global operators? If it's a copied CSV, I'd expect the Subscription to be in a different namespace.

Reply:
In that case, we first need to fetch the CSV then the subscription, and based on `olm.operatorNamespace` annotation adjust the above namespace value.